lib/discourse_automation/scriptable.rb

```diff
@@ -251,12 +252,22 @@ def self.send_pm(
             end
           end
 
+          if pm[:target_emails].present?
+            valid_emails = pm[:target_emails].select { |email| Email.is_valid?(email) }
+            if valid_emails.length != pm[:target_emails].length
+              non_existing_targets += pm[:target_emails] - valid_emails
+              pm[:target_emails] = valid_emails
+            end
+          end
+
           post_created = false
           pm = pm.merge(archetype: Archetype.private_message)
           pm[:target_usernames] = (pm[:target_usernames] || []).join(",")
```

Comment:
Maybe above if you do `pm[:target_usernames] = Array.wrap(pm[:target_usernames])` for all of these above you can simplify these joins too